bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/TestRpcUtilTest.scala

```diff
@@ -120,31 +128,95 @@ class TestRpcUtilTest extends AsyncFlatSpec with BeforeAndAfterAll {
     }
   }
 
-  it should "be able to create a connected node pair with 100 blocks and then delete them" in {
-    BitcoindRpcTestUtil.createNodePair().flatMap {
-      case (client1, client2) =>
+  it should "be able to create a connected node pair with more than 100 blocks and then delete them" in {
+    for {
+      (client1, client2) <- BitcoindRpcTestUtil.createNodePair()
+      _ = {
         assert(client1.getDaemon.authCredentials.datadir.isDirectory)
         assert(client2.getDaemon.authCredentials.datadir.isDirectory)
+      }
+      nodes <- client1.getAddedNodeInfo(client2.getDaemon.uri)
+      _ = {
+        assert(nodes.nonEmpty)
+      }
+      count1 <- client1.getBlockCount
+      count2 <- client2.getBlockCount
+      _ = {
+        assert(count1 > 100)
+        assert(count2 > 100)
+      }
+      _ <- BitcoindRpcTestUtil.deleteNodePair(client1, client2)
+    } yield {
```

Comment:
Is there any reason we can't move these `assert` statements out of the `for` body to make it more readable?

Reply:
Same applies elsewhere

Reply:
Actually looking around more, I don't mind it when it is just `_ = assert(...)` but with multiline asserts, could we just split it up so that instead of having
```
_ = {
  assert(count1 > 100)
  assert(count2 > 100)
}
```
we can have
```
_ = assert(count1 > 100)
_ = assert(count2 > 100)
```